lib/ansible/modules/network/nxos/nxos_udld_interface.py

```diff
@@ -129,6 +130,24 @@ def execute_show_command(command, module, command_type='cli_show'):
     return body
 
 
+def check_load_config(module, config, return_error=False):
+
+    rc, out, err = exec_command(module, 'configure')
```

Comment:
Why `exec_command()` is used here?
Same can be achieved using nxos cliconf plugin `edit_config()` in `cliconf/nxos.py`?

Reply:
edit_config doesn't return anything. Also we do not get return value. exec_command is essential here. Same goes for nxos_install_os module. 
https://github.com/ansible/ansible/pull/31524/files#diff-e34a6a23effc3805a2a2a8a31a13d715R305 `msgs = check_load_config(module, cmds, True)`.
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/nxos/nxos_install_os.py#L441 `data = parse_show_install(load_config(module, commands, True, opts))`.

Reply:
In that case i thin we need change `edit_config()` cliconf api to return responses as well.
IMO `exec_command()` is made available for backward compatibility and should be removed after the transition to cliconf is complete.